cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinderTest.java

```diff
@@ -30,36 +30,50 @@
   public static final String CLUSTER_ID = "cluster_id";
+  private static final double DELTA = 0.01;
 
   @Test
   public void testAnomalyDetection() throws InterruptedException, ExecutionException, TimeoutException {
     KafkaCruiseControl mockKafkaCruiseControl = mockKafkaCruiseControl();
     AdminClient mockAdminClient = mockAdminClient((short) 1);
     TopicReplicationFactorAnomalyFinder anomalyFinder = new TopicReplicationFactorAnomalyFinder(mockKafkaCruiseControl,
                                                                                                 TARGET_TOPIC_REPLICATION_FACTOR,
+                                                                                                TOPIC_REPLICATION_FACTOR_MARGIN,
                                                                                                 mockAdminClient);
     Set<TopicAnomaly> topicAnomalies = anomalyFinder.topicAnomalies();
     assertEquals(1, topicAnomalies.size());
     EasyMock.verify(mockKafkaCruiseControl, mockAdminClient);
   }
 
   @Test
-  public void testSkipTopicWithLargeMinISR() throws InterruptedException, ExecutionException, TimeoutException {
+  public void testAdjustTopicWithLargeMinISR() throws InterruptedException, ExecutionException, TimeoutException {
     KafkaCruiseControl mockKafkaCruiseControl = mockKafkaCruiseControl();
-    AdminClient mockAdminClient = mockAdminClient((short) 2);
+    short expectedMinISR = 2;
+    AdminClient mockAdminClient = mockAdminClient(expectedMinISR);
     TopicReplicationFactorAnomalyFinder anomalyFinder = new TopicReplicationFactorAnomalyFinder(mockKafkaCruiseControl,
                                                                                                 TARGET_TOPIC_REPLICATION_FACTOR,
+                                                                                                TOPIC_REPLICATION_FACTOR_MARGIN,
                                                                                                 mockAdminClient);
     Set<TopicAnomaly> topicAnomalies = anomalyFinder.topicAnomalies();
-    assertTrue(topicAnomalies.isEmpty());
+    assertEquals(1, topicAnomalies.size());
+    TopicReplicationFactorAnomaly topicReplicationFactorAnomaly = (TopicReplicationFactorAnomaly) topicAnomalies.iterator().next();
+    assertEquals(1, topicReplicationFactorAnomaly.getBadTopicsByReplicationFactor().size());
+    // We expect the desired replication factor to be 4 (i.e. TOPIC_REPLICATION_FACTOR_MARGIN + expectedMinISR)
+    assertEquals(TOPIC, topicReplicationFactorAnomaly.getBadTopicsByReplicationFactor()
+                                                     .get((short) (TOPIC_REPLICATION_FACTOR_MARGIN + expectedMinISR))
+                                                     .iterator().next().topicName());
+    // We expect 1 out of 2 partitions of the topic to violate the target RF.
+    assertEquals(0.5, topicReplicationFactorAnomaly.getBadTopicsByReplicationFactor()
```

Comment:
Not really related to this PR, but I would prefer `getBadTopicsByReplicationFactor()` and friends to be called `getBadTopicsByDesiredRF()`.